src/pages/groupComparison/GroupComparisonStore.ts

```diff
@@ -664,18 +654,29 @@ export default class GroupComparisonStore {
     }
 
     readonly mRNAEnrichmentData = makeEnrichmentDataPromise({
-        await:()=>[this.enrichmentsGroup1, this.enrichmentsGroup2,this.mRNAEnrichmentProfile],
+        await:()=>[this.mRNAEnrichmentProfile, this._activeGroupsOverlapRemoved],
         getSelectedProfile:()=>this.mRNAEnrichmentProfile.result,// returns an empty array if the selected study doesn't have any mRNA profiles
         fetchData:()=>{
             // assumes single study for now
-            if (this.enrichmentsGroup1.result && this.enrichmentsGroup2.result && this.mRNAEnrichmentProfile.result) {
+            if (this.mRNAEnrichmentProfile.result) {
+                const molecularProfileId = this.mRNAEnrichmentProfile.result!.molecularProfileId;
+                const groups: MolecularProfileCasesGroupFilter[] = _.map(this._activeGroupsOverlapRemoved.result, group => {
+                    const molecularProfileCaseIdentifiers = _.flatMap(group.studies, study => {
+                        return _.map(study.samples, sampleId => {
+                            return {
+                                caseId: sampleId,
+                                molecularProfileId: molecularProfileId
+                            }
+                        });
+                    });
+                    return {
+                        name: group.nameWithOrdinal,
+                        molecularProfileCaseIdentifiers
+                    }
+                });
                 return internalClient.fetchExpressionEnrichmentsUsingPOST({
```

Comment:
you should break here and return empty if groups.length < 2

Reply:
same for proteinEnrichmentData

Reply:
it won't be invoked if groups.length < 2

https://github.com/cBioPortal/cbioportal-frontend/blob/4049894b6b08d52e6d1f04c7794360d9a110f8c7/src/pages/groupComparison/GroupComparisonUtils.tsx#L321